sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityCredential.java

```diff
@@ -53,22 +58,19 @@
      * @return the client ID of user assigned or system assigned identity.
      */
     public String getClientId() {
-        return this.appServiceMSICredential != null
-            ? this.appServiceMSICredential.getClientId()
-            : this.virtualMachineMSICredential.getClientId();
+        return managedIdentityServiceCredential.getClientId();
```

Comment:
This can potentially throw NPE if `managedIdentityServiceCredential` is null. So, as mentioned above, it might be better to just throw an exception in ctor itself if it's null.

Reply:
udpated